### ISSUE TITLE
Wrong error message on extended interfaces

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#81]
```php
interface A {}
interface B extends A {}
function foo(A $a) : B {
  return $a;
}
```
Expecting `MoreSpecificReturnType`, got `InvalidReturnType`
